lib/rubocop/cop/rspec/change_without_expect.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for `change` matcher usage without an `expect` block.
+      #
+      # This cop only considered a matcher if it is chained with
+      # `from`, `by`, `by_at_least`, or `by_at_most`.
+      #
+      # @example
+      #   # bad
+      #   it 'changes the count' do
+      #     change(Counter, :count).by(1)
+      #   end
+      #
+      #   # bad
+      #   it 'changes the count' do
+      #     change(Counter, :count).by_at_least(1)
+      #   end
+      #
+      #   # bad
+      #   it 'changes the count' do
+      #     change(Counter, :count).by_at_most(1)
+      #   end
+      #
+      #   # bad
+      #   it 'changes the count' do
+      #     change(Counter, :count).from(0)
+      #   end
+      #
+      #   # bad
+      #   it 'changes the count' do
+      #     change(Counter, :count).from(0).to(1)
+      #   end
+      #
+      #   # good - no chained matchers
+      #   it 'changes the count' do
+      #     change(Counter, :count)
+      #   end
+      #
+      #   # good - ignore not change matcher
+      #   it 'changes the count' do
+      #     change(Counter, :count).from(0).by(1)
+      #   end
+      #
+      #   # good - within expect block
+      #   it 'changes the count' do
+      #     expect { subject }.to change(Counter, :count).by(1)
+      #   end
+      #
+      class ChangeWithoutExpect < RuboCop::Cop::Base
+        include RangeHelp
+
+        MSG = 'Use `change` matcher within an `expect` block.'
+        RESTRICT_ON_SEND = [:change].freeze
+        SINGLE_RESTRICTED_METHODS = %i[by by_at_least by_at_most from].freeze
```

Comment:
There’s also “to” https://github.com/rspec/rspec/blob/d064fd0e9f00af834dfcfbe55e74b066a76e98e6/rspec-expectations/lib/rspec/matchers/built_in/change.rb#L36